digdag-standards/src/main/java/io/digdag/standards/operator/aws/EmrOperatorFactory.java

```diff
@@ -0,0 +1,1739 @@
+                String key = (baseKey != null ? baseKey : "") + prefix + "/" + reference.filename();
+                builder.s3Uri(new AmazonS3URI("s3://" + staging.get().getBucket() + "/" + key));
+            }
+            else {
+                builder.s3Uri(new AmazonS3URI(reference.reference().get()));
+            }
+
+            RemoteFile remoteFile = builder.build();
+
+            if (reference.local()) {
+                files.add(ImmutableStagingFile.builder()
+                        .template(template)
+                        .file(remoteFile)
+                        .build());
+            }
+
+            return remoteFile;
+        }
+
+        void stageFiles()
+        {
+            if (files.isEmpty()) {
+                return;
+            }
+
+            TransferManager transferManager = new TransferManager(s3);
+            List<PutObjectRequest> requests = new ArrayList<>();
+
+            for (StagingFile f : files) {
+                logger.info("Staging {} -> {}", f.file().reference().filename(), f.file().s3Uri());
+                requests.add(stagingFilePutRequest(f));
+            }
+
+            try {
+                List<Upload> uploads = requests.stream()
+                        .map(transferManager::upload)
+                        .collect(toList());
+
+                for (Upload upload : uploads) {
+                    try {
+                        upload.waitForCompletion();
+                    }
+                    catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                        throw new TaskExecutionException(e, TaskExecutionException.buildExceptionErrorConfig(e));
+                    }
+                }
+            }
+            finally {
+                transferManager.shutdownNow(false);
+                requests.forEach(r -> closeQuietly(r.getInputStream()));
+            }
+        }
+
+        private PutObjectRequest stagingFilePutRequest(StagingFile file)
+        {
+            AmazonS3URI uri = file.file().s3Uri();
+            FileReference reference = file.file().reference();
+
+            switch (reference.type()) {
+                case LOCAL: {
+                    if (file.template()) {
+                        try {
+                            String content = workspace.templateFile(templateEngine, reference.filename(), UTF_8, params);
+                            byte[] bytes = content.getBytes(UTF_8);
+                            ObjectMetadata metadata = new ObjectMetadata();
+                            metadata.setContentLength(bytes.length);
+                            return new PutObjectRequest(uri.getBucket(), uri.getKey(), new ByteArrayInputStream(bytes), metadata);
+                        }
+                        catch (IOException | TemplateException e) {
+                            throw new TaskExecutionException(e, TaskExecutionException.buildExceptionErrorConfig(e));
```

Comment:
In case of TemplateException, it sounds there is something wrong with configuration and we'd better use ConfigException instead?